src/Action/Iterator.php

```diff
@@ -33,71 +37,175 @@ public function __construct(array $data)
+     *
+     * @return bool
      */
-    public function like($field, $value)
+    protected function match(array $row, AbstractScope $scope)
     {
-        $this->generator = new \CallbackFilterIterator($this->generator, function ($row) use ($field, $value) {
-            // skip row. does not have field at all
-            if (!array_key_exists($field, $row)) {
-                return false;
+        $match = false;
+
+        // simple condition
+        if ($scope instanceof Condition) {
+            $args = $scope->toArray();
+
+            $field = $args[0];
+            $operator = $args[1] ?? null;
+            $value = $args[2] ?? null;
+            if (count($args) == 2) {
+                $value = $operator;
+
+                $operator = '=';
             }
 
-            $fieldValStr = (string) $row[$field];
+            if (!is_a($field, Field::class)) {
+                throw (new Exception('Persistence\Array_ driver condition unsupported format'))
+                    ->addMoreInfo('reason', 'Unsupported object instance ' . get_class($field))
+                    ->addMoreInfo('condition', $scope);
+            }
 
-            $value = trim($value);
-            $clean_value = trim($value, '%');
-            // the row field exists check the position of the "%"(s)
-            switch ($value) {
-                // case "%str%"
-                case substr($value, -1, 1) === '%' && substr($value, 0, 1) === '%':
-                    return strpos($fieldValStr, $clean_value) !== false;
+            if (isset($row[$field->short_name])) {
+                $match = $this->where($row[$field->short_name], $operator, $value);
+            }
+        }
 
-                    break;
-                // case "str%"
-                case substr($value, -1, 1) === '%':
-                    return substr($fieldValStr, 0, strlen($clean_value)) === $clean_value;
+        // nested conditions
+        if ($scope instanceof Scope) {
+            $matches = [];
 
-                    break;
-                // case "%str"
-                case substr($value, 0, 1) === '%':
-                    return substr($fieldValStr, -strlen($clean_value)) === $clean_value;
+            foreach ($scope->getActiveComponents() as $component) {
+                $matches[] = $subMatch = (bool) $this->match($row, $component);
 
+                // do not check all conditions if any match required
+                if ($scope->isOr() && $subMatch) {
                     break;
-                // full match
-                default:
-                    return $fieldValStr == $clean_value;
+                }
             }
 
-            return false;
-        });
+            // any matches && all matches the same (if all required)
+            $match = array_filter($matches) && ($scope->isAnd() ? count(array_unique($matches)) === 1 : true);
+        }
 
-        return $this;
+        return $match;
+    }
+
+    protected function where($v1, $operator, $v2)
```

Comment:
protected, but should be probably named better, this function does not add conditions, it calculates the outcome

Reply:
done